### PR TITLE
Ensure that `TilingPattern`s have valid (non-zero) /BBox arrays (issue 8330)

### LINKED ISSUE #8330: Part of pdf is not rendering [Invalid canvas size error in console]
Link to PDF file (or attach file here
[FailingPDF.pdf](https://github.com/mozilla/pdf.js/files/954710/FailingPDF.pdf)

Configuration:
- Chrome (57.0.2987.133)
- MAC OS (10.10.5)
- PDF.js version:1.8.252
- Is an extension: No

Steps to reproduce the problem:
1. Kindly upload the file attached above in the https://mozilla.github.io/pdf.js/web/viewer.html
2. When the first page is rendered it throw an error on the console telling invalid canvas size

A part of the pdf present in the first page was not loaded, and the loading is stuck.
Below is the error, and the loading sign on the pdf first page
error
loadingsign

Ideally it should have loaded the part of the pdf.

### PR DESCRIPTION
**Pull request by Snuffleupagus**
Fixes #8330.

*Please note:* This is a tentative patch, since I'm not entirely certain if it's the best solution; but it does work!

### CODE REVIEW

**Comment by Snuffleupagus** [comment]
*Checking if the test "failures" in Chrome on Windows is intermittent.*

/botio-windows test

File: src/core/evaluator.js [hunk 1]
```diff
@@ -480,30 +480,32 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
...
-        }, patternDict, args));
+          argsArray: tilingOpList.argsArray,
+        }, patternDict, args);
+      }).then(function(tilingPatternIR) {
+        // Add the dependencies to the parent operator list so they are resolved
+        // before the sub operator list is executed synchronously.
+        operatorList.addDependencies(tilingOpList.dependencies);
+        operatorList.addOp(fn, tilingPatternIR);
+      }, function(reason) {
+        warn(`handleTilingType - ignoring pattern: "${reason}".`);
```
Comment (Snuffleupagus) [inline src/core/evaluator.js]: I'm not sure if we should send an `UnsupportedFeature` notification here, or if it doesn't matter!?

File: src/core/pattern.js [hunk 2]
```diff
@@ -795,12 +795,17 @@ Shadings.Dummy = (function DummyClosure() {
...
+  let matrix = dict.getArray('Matrix');
+  let bbox = Util.normalizeRect(dict.getArray('BBox'));
+  let xstep = dict.get('XStep');
+  let ystep = dict.get('YStep');
+  let paintType = dict.get('PaintType');
+  let tilingType = dict.get('TilingType');
+
+  // Ensure that the pattern has a non-zero width and height, to prevent errors
+  // in `pattern_helper.js` (fixes issue8330.pdf).
+  assert((bbox[2] - bbox[0]) !== 0 && (bbox[3] - bbox[1]) !== 0,
```
Comment (yurydelendik) [inline src/core/pattern.js]: (it's pre-existing practice but) let's change this to `if..throw new Error`. I'll will be opening issue to re-inspect assert usage (see #8506)

**Review by yurydelendik — APPROVED** [review]
Looks good with change below.

File: src/core/pattern.js (hunk 2 above)
Comment (Snuffleupagus) [inline src/core/pattern.js]: Sure, I've updated the patch; thank you for the review!

**Pull request merged**